### DISCUSSION TITLE
Severity

### DISCUSSION THREAD

**Discussion by spaelling** [maester365/maester#535] [Ideas]
The Maester test results can be quite overwhelming, especially for novices. One of the major decisions is where to start (with failed tests), and this is where the concept of severity comes into play.

Optimally this is an entirely separate function along with category and tags, but to make this trivial to implement we could use the `tag` functionality. I would suggest simply adding one of the following tags to each Maester test

- Severity:Critical
- Severity:High
- Severity:Medium
- Severity:Low

This also means that we can utilize the existing functionality of running tests with a specific tag.

```powershell
Invoke-Maester -Tag "Severity:High"
```

Which allows us to create a report with the highest severity tests first, and then work our way down.

**Comment by simon-barratt-seq** [comment]
I think this is a great idea

**Comment by merill** [comment]
The criticality should come ideally come from the test itself but I agree that being able to filter and run just the required tests would be neat as well.

Will need to find a way to keep them in sync. For now, the only way I can think of doing this is to add a regression test that compares the output vs the tag and call out if they are different.
